model2vec/train/classifier.py

```diff
@@ -0,0 +1,293 @@
+            trainer = pl.Trainer(
+                max_epochs=500,
+                callbacks=callbacks,
+                val_check_interval=val_check_interval,
+                check_val_every_n_epoch=check_val_every_epoch,
+                accelerator=device,
+                default_root_dir=tempdir,
+            )
+
+            trainer.fit(
+                c,
+                train_dataloaders=train_dataset.to_dataloader(shuffle=True, batch_size=batch_size),
+                val_dataloaders=val_dataset.to_dataloader(shuffle=False, batch_size=batch_size),
+            )
+            best_model_path = trainer.checkpoint_callback.best_model_path  # type: ignore
+            best_model_weights = torch.load(best_model_path, weights_only=True)
+
+        state_dict = {}
+        for weight_name, weight in best_model_weights["state_dict"].items():
+            state_dict[weight_name.removeprefix("model.")] = weight
+
+        self.load_state_dict(state_dict)
+        self.eval()
+
+        return self
+
+    def _initialize(self, y: list[str]) -> None:
+        """Sets the out dimensionality, the classes and initializes the head."""
+        classes = sorted(set(y))
+        self.classes_ = classes
+
+        if len(self.classes) != self.out_dim:
+            self.out_dim = len(self.classes)
+
+        self.head = self.construct_head()
+        self.embeddings = nn.Embedding.from_pretrained(self.vectors.clone(), freeze=False, padding_idx=self.pad_id)
+        self.w = self.construct_weights()
+        self.train()
+
+    def _prepare_dataset(self, X: list[str], y: list[str], max_length: int = 512) -> TextDataset:
+        """Prepare a dataset."""
+        # This is a speed optimization.
+        # assumes a mean token length of 10, which is really high, so safe.
+        truncate_length = max_length * 10
+        X = [x[:truncate_length] for x in X]
+        tokenized: list[list[int]] = [
+            encoding.ids[:max_length] for encoding in self.tokenizer.encode_batch_fast(X, add_special_tokens=False)
+        ]
+        labels_tensor = torch.Tensor([self.classes.index(label) for label in y]).long()
+
+        return TextDataset(tokenized, labels_tensor)
+
+    @staticmethod
+    def _train_test_split(
+        X: list[str], y: list[str], test_size: float
+    ) -> tuple[list[str], list[str], list[str], list[str]]:
+        """Split the data."""
+        label_counts = Counter(y)
+        if min(label_counts.values()) < 2:
+            logger.info("Some classes have less than 2 samples. Stratification is disabled.")
+            return train_test_split(X, y, test_size=test_size, random_state=42, shuffle=True)
+        return train_test_split(X, y, test_size=test_size, random_state=42, shuffle=True, stratify=y)
+
+    def to_pipeline(self) -> StaticModelPipeline:
+        """Convert the model to an sklearn pipeline."""
+        static_model = self.to_static_model()
+
+        random_state = np.random.RandomState(42)
```

Comment:
same w.r.t. random seed. If we don't want users to be able to define this. Perhaps we could create `RANDOM_SEED` somewhere and re-use it instead of copy-pasting 42?